front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsControllerV2.groovy

```diff
@@ -0,0 +1,163 @@
+  @PostFilter("hasPermission(filterObject.name, 'APPLICATION', 'READ')")
+  @ApiOperation(value = "", notes = """Fetch all applications.
+
+    Supports filtering by one or more attributes:
+    - ?email=my@email.com
+    - ?email=my@email.com&name=flex""")
+  @RequestMapping(method = RequestMethod.GET)
+  Set<Application> applications(@RequestParam Map<String, String> params) {
+    if (params.isEmpty()) {
+      return applicationDAO.all()
+    }
+
+    return applicationDAO.search(params)
+  }
+
+  @PreAuthorize("hasPermission(#app.name, 'APPLICATION', 'CREATE')")
+  @ApiOperation(value = "", notes = "Create an application")
+  @RequestMapping(method = RequestMethod.POST, value = "")
+  Application create(@RequestBody final Application app) {
+    return getApplication().initialize(app).withName(app.getName()).save()
+  }
+
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'WRITE')")
+  @ApiOperation(value = "", notes = "Delete an application")
+  @RequestMapping(method = RequestMethod.DELETE, value = "/{application}")
+  void delete(@PathVariable String application, HttpServletResponse response) {
+    getApplication().initialize(new Application().withName(application)).delete()
+    response.setStatus(HttpStatus.ACCEPTED.value())
+  }
+
+  @PreAuthorize("hasPermission(#app.name, 'APPLICATION', 'WRITE')")
+  @ApiOperation(value = "", notes = "Update an existing application")
+  @RequestMapping(method = RequestMethod.PUT, value = "/{name}")
+  Application update(@PathVariable String name, @RequestBody final Application app) {
+    if (name.trim() != app.getName()) {
+      log.error("non matching application name {}", name)
+      throw new InvalidApplicationRequestException("provided application name does not match")
+    }
+
+    def application = getApplication()
+    Application existingApplication = application.findByName(app.getName())
+    application.initialize(existingApplication).withName(app.getName()).update(app)
+    return app
+  }
+
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'READ')")
+  @ApiOperation(value = "", notes = "Fetch a single application by name")
+  @RequestMapping(method = RequestMethod.GET, value = "/{application}")
+  Application get(@PathVariable final String application) {
+    if (!application?.trim()) {
+      throw new NotFoundException("No application name provided")
+    }
+
+    return applicationDAO.findByName(application.toUpperCase())
+  }
+
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'READ')")
+  @RequestMapping(value = '{application}/history', method = RequestMethod.GET)
+  Collection<Application> getHistory(@PathVariable String application,
+                                     @RequestParam(value = "limit", defaultValue = "20") int limit) {
+    return applicationDAO.getApplicationHistory(application, limit)
+  }
+
+  @PreAuthorize("@fiatPermissionEvaluator.isAdmin()")
+  @RequestMapping(method = RequestMethod.POST, value = "/batchUpdate")
```

Comment:
@duftler and I threw around the idea of a `/admin` endpoint, where all of these batch & migration endpoints could live.